### PR TITLE
Chore: notify on CI periodic run failures in Zulip

### PR DESCRIPTION
**Pull request by andrei-near**
Screenshot 2024-12-10 at 14 35 09

### CODE REVIEW

File: .github/workflows/ci_nightly.yml [hunk 1]
```diff
@@ -36,6 +36,20 @@ jobs:
       - uses: taiki-e/install-action@9b5b983efc779f85e5e5d11539f005e85ccb27ff
         with:
           tool: just,cargo-nextest
-
       # Run the tests:
-      - run: just nextest-slow ${{ matrix.type }}
+      - uses: mathiasvr/command-output@v2.0.0
```
Comment (nagisa) [inline .github/workflows/ci_nightly.yml:40]: Lets always pin a specific commit :) Much harder to find a hash collision than to force push a tag.

File: .github/workflows/ci_nightly.yml [hunk 2]
```diff
@@ -36,6 +36,20 @@ jobs:
...
+        with:
+          run: just nextest-slow ${{ matrix.type }}
+        id: run_nextest
+
+      - name: Notify in Zulip about failures
+        uses: zulip/github-actions-zulip/send-message@v1
```
Comment (nagisa) [inline .github/workflows/ci_nightly.yml:46]: Similarly here.

**Review by VanBarbascu — COMMENTED** [review]
What will be the format of the message? How big is `steps.run_nextest.outputs.stderr`?

File: .github/workflows/ci_nightly.yml (hunk 1 above)
Comment (andrei-near) [reply .github/workflows/ci_nightly.yml]: done

File: .github/workflows/ci_nightly.yml (hunk 2 above)
Comment (andrei-near) [reply .github/workflows/ci_nightly.yml]: done

**Comment by andrei-near** [comment]
@VanBarbascu step output max size is 1MB which is roughly 1,048,576 characters. The step will only report stderr and everything over 1M will get truncated. Either way, 1MB is enough to give us a rough idea of whats wrong and we can review markdown to something like spoiler if the output will be too noisy

**Review by nagisa — APPROVED** [review]

**Pull request merged**